test/IISIntegration.FunctionalTests/Utilities/IISApplication.cs

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.AspNetCore.Server.IntegrationTesting.Common;
+using Microsoft.Extensions.Logging;
+using Microsoft.Web.Administration;
+
+namespace Microsoft.AspNetCore.Server.IntegrationTesting
+{
+    internal class IISApplication
+    {
+        private readonly ServerManager _serverManager = new ServerManager();
+        private readonly DeploymentParameters _deploymentParameters;
+        private readonly ILogger _logger;
+        private string _originalAspNetCoreDllLocation = "";
+
+        public IISApplication(DeploymentParameters deploymentParameters, ILogger logger)
+        {
+            _deploymentParameters = deploymentParameters;
+            _logger = logger;
+            WebSiteName = CreateTestSiteName();
+            AppPoolName = $"{WebSiteName}Pool";
+        }
+
+        public string WebSiteName { get; }
+
+        public string AppPoolName { get; }
+
+        public void StartIIS(Uri uri, string contentRoot)
+        {
+            using (_logger.BeginScope("StartIIS"))
+            {
+                var port = uri.Port;
+                if (port == 0)
+                {
+                    port = (uri.Scheme == "https") ? TestPortHelper.GetNextSSLPort() : TestPortHelper.GetNextPort();
```

Comment:
This SslPort logic won't work, those ports are specifically bound to IIS Express. You'd have to create an ACL using the dev cert. Throw NotSupported for now?